mathics/builtin/numbers/calculus.py

```diff
@@ -228,9 +228,9 @@ def apply(self, f, x, evaluation):
         elif f == x:
             return Integer1
         elif isinstance(f, Atom):  # Shouldn't happen
-            1 / 0
```

Comment:
Not sure even if the `elif` block is necessary.  That `f` is not an atom should be handled by the first `if` clause where we check that `f` is free of the derivative parameter.

Reply:
Ok - thanks. All of this has been rewritten in 16ad3908